o3de.sh script does not understand tilde

**Describe the bug**
o3de.sh script does not understand tilde character

**Assets required**
No assets

**Steps to reproduce**
Steps to reproduce the behavior:
1. Install o3de deb package
2. Call script with tilde character
3. See error

```
/opt/O3DE/26.05/scripts/o3de.sh register --all-gems-path ~/devroot/2605/o3de-extras/Gems/
1.11.1
[INFO] o3de.register: O3DE Manifest file location: `/home/jhanca/.o3de/o3de_manifest.json`
[ERROR] o3de.register: gem path is not a dir.
[INFO] o3de: Completed with issues: result 1


/opt/O3DE/26.05/scripts/o3de.sh register --all-gems-path /home/jhanca/devroot/2505/o3de-extras/Gems/
1.11.1
[INFO] o3de.register: O3DE Manifest file location: `/home/jhanca/.o3de/o3de_manifest.json`
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/XR` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/WarehouseAutomation` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/WarehouseAssets` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/SimulationInterfaces` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/ROS2SampleRobots` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/ROS2` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/OpenXRVk` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/MachineLearning` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/LevelGeoreferencing` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/AzQtComponentsForPython` to key `external_subdirectories` in the manifest.
[INFO] o3de.register: Adding `/home/jhanca/devroot/2505/o3de-extras/Gems/AudioEngineWwise` to key `external_subdirectories` in the manifest.
[INFO] o3de: Success!
```

**Expected behavior**
Tilde character is converted into path of the home folder

**Actual behavior**
Path is not recognized

**Screenshots/Video**
If applicable, add screenshots and/or a video to help explain your problem.

**Found in Branch**
Stabilization deb package pulled Apr, 02 2026

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
Please provide the SHA/hash that identifies the latest commit from the o3de/o3de repository if the issue is reproducible in the development or other official branches. 
You can get the commit ID by running the `git rev-parse HEAD` command on your current branch.

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Ubuntu Linux
 - Version 24.04.4 LTS
 - CPU i7-12700H
 - GPU RTX 3070 Laptop
 - Memory 32 GB

**Additional context**
There are some more errors when running o3de for the first time:
```
Application: 
==================================================================
Application: Trace::Warning
 /data/workspace/o3de/Code/Framework/AzFramework/AzFramework/Application/Application.cpp(674): 'void AzFramework::CreateUserCache(const AZ::IO::FixedMaxPath &, AZ::IO::FileIOBase &)'
Application: Couldn't find a valid asset cache folder after 128 attempts. Setting cache folder to /opt/O3DE/26.05/bin/Linux/profile/Default/user/Cache

Application: ==================================================================
System: Network layer initialized
python: 
==================================================================
python: Trace::Error
 /data/workspace/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/API/PythonLoader.cpp(111): 'static AZ::IO::FixedMaxPath AzToolsFramework::EmbeddedPython::PythonLoader::GetPythonExecutablePath(AZ::IO::PathView, const char *)'
python: Missing python venv file at /home/jhanca/.o3de/Python/venv/c8beaf43/pyvenv.cfg. Make sure to run python/get_python.
python: ==================================================================
python: 
==================================================================
python: Trace::Error
 /data/workspace/o3de/Code/Tools/ProjectManager/Source/PythonBindings.cpp(290): 'virtual bool O3DE::ProjectManager::PythonBindings::StartPython()'
python: Python home path does not exist: 
python: ==================================================================

```

--- a/scripts/o3de.sh
+++ b/scripts/o3de.sh
@@ -37,5 +37,5 @@ if [ ! -f "$PYTHON_EXECUTABLE" ]; then
 fi
 
 #run the o3de.py pass along the command
-$PYTHON_EXECUTABLE "$SCRIPT_DIR/o3de.py" $*
+$PYTHON_EXECUTABLE "$SCRIPT_DIR/o3de.py" "$@"
 exit $?

